From da11af5b6c2ae16e90ff8fb0482d8c7537a66c62 Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Tue, 4 Jun 2013 15:21:00 +0200
Subject: [PORT FROM MAIN] crashtool: use logcat for logging

BZ: 113505

Remove old logging utilities; include logcat's ones.

Change-Id: Iec125ee3cd63b9cc7e8dd33f1113972de2346b86
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/111611
Reviewed-by: Haslam, AxelX <axelx.haslam@intel.com>
Reviewed-by: Isaksen, FrodeX <frodex.isaksen@intel.com>
Reviewed-by: Lestienne, ChristopheX <christophex.lestienne@intel.com>
Reviewed-by: Fiat, Christophe <christophe.fiat@intel.com>
Tested-by: Lestienne, ChristopheX <christophex.lestienne@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libc/include/ctm.h             |   51 ----------------------------------------
 libc/include/lct.h             |    2 +-
 libc/kernel/common/linux/kct.h |    1 +
 3 files changed, 2 insertions(+), 52 deletions(-)
 delete mode 100644 libc/include/ctm.h

diff --git a/libc/include/ctm.h b/libc/include/ctm.h
deleted file mode 100644
index b9cd7f1..0000000
--- a/libc/include/ctm.h
+++ /dev/null
@@ -1,51 +0,0 @@
-#ifndef CTM_H_
-#  define CTM_H_
-
-#  include <sys/socket.h>
-#  include "linux/kct.h"
-#  include <stdint.h>
-
-#  define DEBUG
-#  ifdef DEBUG
-#    define debug(Args...) fprintf(stderr, ## Args)
-#  else
-#    define debug(Args...) { }
-#  endif
-
-#  define info(Args...) fprintf(stdout, ## Args)
-#  define warning(Args...) fprintf(stderr, ## Args)
-#  define error(Args...) fprintf(stderr, ## Args)
-
-#  define max(X, Y)  ((X) < (Y) ? (Y) : (X))
-
-extern int thread_should_stop;
-extern int sock_nl_fd;
-extern int sock_fd;
-
-enum ctm_ev_pending {
-	EV_PENDING_ALL,
-	EV_PENDING_NLSOCKET,
-	EV_PENDING_USOCKET,
-	EV_PENDING_NONE
-};
-
-/* ctm_netlink.c */
-extern int ctm_nl_init(void);
-extern int ctm_nl_sendto_kct(int fd, int type, const void *data, unsigned int size);
-extern struct kct_packet *ctm_nl_get_packet(int fd);
-
-/* ctm_comm.c */
-extern enum ctm_ev_pending	ctm_comm_wait_event(long timeout);
-extern int				lct_server_init(void);
-extern void			ctm_comm_handle_socket_msg(void);
-extern void			ctm_comm_handle_netlink_msg(void);
-
-/* ctm_print.c */
-extern void print_ct_attchmt(struct ct_attchmt *at);
-extern void print_all_attchmnt(struct ct_event *ev);
-extern void print_ct_event(struct ct_event *ev);
-
-/* ctm_handlers.c */
-extern int ct_handler(struct ct_event *ev, char *dir, char *dsuffix, char *tsuffix);
-
-#endif /* !CTM_H_ */
diff --git a/libc/include/lct.h b/libc/include/lct.h
index 4ca08d4..a0cac9e 100644
--- a/libc/include/lct.h
+++ b/libc/include/lct.h
@@ -13,7 +13,7 @@
 #include <linux/kct.h>
 
 /* prefix of properties used to filter events */
-#define	PROP_PREFIX "dev.log."
+#define	PROP_PREFIX	"dev.log"
 
 #define PKT_SIZE(Ev) (((struct ct_event*)(Ev))->attchmt_size + sizeof(struct ct_event))
 #define SK_NAME "\0crashtool_socket"
diff --git a/libc/kernel/common/linux/kct.h b/libc/kernel/common/linux/kct.h
index c172474..479a395 100644
--- a/libc/kernel/common/linux/kct.h
+++ b/libc/kernel/common/linux/kct.h
@@ -1,6 +1,7 @@
 #ifndef KCT_H_
 #  define KCT_H_
 
+#  include <sys/socket.h>
 #  include <linux/netlink.h>
 
 /* flags to optionally filter events on android property activation */
-- 
1.7.4.1

